controller/controller_server/csi_controller_server.py

```diff
@@ -246,14 +248,20 @@ def DeleteVolume(self, request, context):
                     logger.debug("Deleting volume {0}".format(vol_id))
                     array_mediator.delete_volume(vol_id)
 
-                except controller_errors.VolumeNotFoundError as ex:
+                except controller_errors.ObjectNotFoundError as ex:
                     logger.debug("volume was not found during deletion: {0}".format(ex))
 
                 except controller_errors.PermissionDeniedError as ex:
                     context.set_code(grpc.StatusCode.PERMISSION_DENIED)
                     context.set_details(ex)
                     return csi_pb2.DeleteVolumeResponse()
 
+        except controller_errors.ObjectIsStillInUseError:
```

Comment:
if we want this addition - then why should we keep `is_volume_has_snapshots`?

Reply:
because volume can have clone that he is still write to..

Reply:
I still couldn't understand why we need both (and why `snapshots`)?

Reply:
done

Reply:
so where is `ObjectIsStillInUseError` being raised by the mediators `delete_volume`?